server/lib/enrich/article.js

```diff
@@ -13,6 +13,34 @@ const {
 const RE_BAD_CHARS = /[^A-Za-z0-9_]/gm;
 const RE_SPACE = /\s/gm;
 
+function extractFourishEmbeds(contentHTMLBody) {
+	try {
+		// Regex to extract Flourish IDs
+		const flourishIdRegex = /data-flourish-id="(\d+)"/g;
+
+		const flourishEmbeds = [];
+		let match;
+
+		while ((match = flourishIdRegex.exec(contentHTMLBody)) !== null) {
```

Comment:
Comment: Regex.exec has a state inside it, explained [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#description). Maybe we can use a simpler approach which doesn't involve any states. 

Example:
```
for (const match of contentHTMLBody.matchAll(regex)) {
  // match[1] will be captured ID I believe
  ids2.push(match[1]);
}
```